scripts/west_commands/runners/bossac.py

```diff
@@ -180,9 +181,12 @@ def magic_delay(self):
 
     def make_bossac_cmd(self):
         self.ensure_output('bin')
-        cmd_flash = [self.bossac, '-p', self.port, '-R', '-e', '-w', '-v',
+        cmd_flash = [self.bossac, '-p', self.port, '-R', '-w', '-v',
```

Comment:
I don't understand why you are using the flash command if you intention is not to flash. On my perspective there is nothing wrong with the erase here and it should always erase the device before flash.

Would be the case to have a different command on west that fits with your needs ?

Reply:
The intention is to flash the program, just not to erase the entire flash of the target. Many of the other flashing tools behave this way, so that flash ranges used by the application to preserve settings don't lose their settings, as you are iterating flashing the code.

If you need a full erase, you can pass the  `--erase` flag.